processing/data_collection/gazette/spiders/sp_valinhos.py

```diff
@@ -0,0 +1,45 @@
+# -*- coding: utf-8 -*-
+import datetime
+import scrapy
+
+from gazette.items import Gazette
+from gazette.spiders.base import BaseGazetteSpider
+
+
+class SpValinhosSpider(BaseGazetteSpider):
+    TERRITORY_ID = "3556206"
+    name = "sp_valinhos"
+    allowed_domains = ["valinhos.sp.gov.br"]
+    base_urls = ["http://www.valinhos.sp.gov.br"]
+    selector_url = "http://www.valinhos.sp.gov.br/boletins?title=&field_boletim_publicacao_value%5Bvalue%5D%5Bmonth%5D={}&field_boletim_publicacao_value%5Bvalue%5D%5Byear%5D={}"
+
+    def parse(self, response):
+        today = datetime.date.today()
+        next_year = today.year + 1
+        for year in range(2015, next_year):
+            for month in range(1, 13):
+                if year == today.year and month > today.month:
+                    return
+
+                url = self.selector_url.format(month, year)
+                yield scrapy.Request(url, self.parse_month_page)
+
+    def parse_month_page(self, response):
+        items = []
+        blocks = response.xpath('//*[@id="block-system-main"]/div/div[2]/div/ul/li')
```

Comment:
As rule of thumb, it is better to have your selectors as simple as possible. This will make maintenance much easier.
If for some reason, a new `<div>` is added, it will will break your selector and you will stop to get the data, even if the page didn't change much.

If you look better at the source code of the page, you will see that what you really want are the `<li>` tags that are inside a `<div>` with class `item-list`.

A much simpler selector (you can use CSS selectors!) would be:
`response.css("div.item-list li")`

If you look better at the content and the classes of the page, you can do even more specific, getting only the `li` tags that contains `views-row` class:

`response.css("li.views-row")` or `response.css(".views-row")`